### PR TITLE
Port standardization for openstack services with backward compatibility for existing keystone port

### PR DESCRIPTION
**Pull request by narengan**
Port variable change

uncommented code

uncommented code

added old all.yml and hosts file

added a role for endpoints

changed hard coded ports to variables

Added endpoint role dependency and cleaned up code

cleaned up code

removed unwanted line

added keystone_user endpoints variable since we arent using the default ports as backend

removed left over code

Update main.yml

Update main.yml

removed code from cinder-common by mistake- fixing it

### CODE REVIEW

File: roles/endpoints/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      backend_api: 6384                                                                 
+      haproxy_api: 6385
+  swift:
+    name: swift
+    type: object-storage
+    description: 'Object Storage Service'
+    version: v1
+    url:
+      admin: https://{{ fqdn }}:8090
+      internal: https://{{ undercloud_floating_ip }}:8090
```
Comment (ryshah) [inline roles/endpoints/defaults/main.yml]: swift internal url should not use undercloud_floating_ip. HAProxy VIP parameter was created to be set to Swift ucarp ip

File: roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+    [ 'cinder', endpoints.cinder.port.backend_api, endpoints.cinder.port.haproxy_api, false ],
   ]
 -%}
-{% if heat.enabled|default('True')|bool -%}
-    {% set _ = haproxy_services.append(['heat', 8004, 8005, false]) %}
-    {% set _ = haproxy_services.append(['heat-cfn', 8000, 8001, false]) %}
-{% endif -%}
-
 {% if magnum.enabled|default('False')|bool -%}
     {% set _ = haproxy_services.append(['magnum', 9512, 9511, false]) %}
```
Comment (j2sol) [inline roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg]: Now that magnum is in, these should be templated as well.

File: roles/swift-proxy/templates/etc/swift/proxy-server.conf [hunk 3]
```diff
@@ -85,8 +85,10 @@ use = egg:swift#dlo
 
 [filter:authtoken]
 paste.filter_factory = keystonemiddleware.auth_token:filter_factory
-identity_uri = https://{{ endpoints.keystone }}:35358/
-auth_uri = https://{{ endpoints.keystone }}:5001/
+identity_uri = {{ endpoints.keystone.url.admin }}/
+auth_uri = {{ endpoints.keystone.url.internal }}/
+auth_host = {{ endpoints.main }}
+auth_port = {{ endpoints.keystone.port.haproxy_api }}
```
Comment (j2sol) [inline roles/swift-proxy/templates/etc/swift/proxy-server.conf]: Is auth_host and auth_port necessary here? I thought keystonemiddleware could get all it needs from identity_uri and auth_uri.

**Comment by j2sol** [comment]
Very close now!

File: roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg (hunk 2 above)
Comment (narengan) [inline roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg]: will be added in the next PR

**Comment by j2sol** [comment]
Okay, looks good now. I'll remove the flag and wait for tests to pass.

**Comment by narengan** [comment]
could the build have failed because I changed all the ports ?

**Comment by narengan** [comment]
retest

**Comment by narengan** [comment]
retest

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
This is now failing to create the default image. Not sure why, needs some investigation.

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
Please rebase on master to pick up required fixes to pass tests.

**Comment by ddaskal** [comment]
retest

**Comment by narengan** [comment]
retest

**Comment by ddaskal** [comment]
retest

**Pull request merged**